### PR TITLE
#79 - Update get/bid endpoint - Aisha Athman Lali

### PR DESCRIPTION
**Pull request by aishaathmanlali**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

**Review by soonick — CHANGES_REQUESTED** [review]
Test your code before sending it for review

**Comment by aishaathmanlali** [comment]
> I tested it now its working after some modifications

File: server/api.js [hunk 1]
```diff
@@ -390,32 +391,62 @@ router.get("/tenders", async (req, res) => {
...
+			totalBidsParams = [userId, tenderID];
+			bidsQuery = `
+				SELECT bid.* FROM bid 
+				JOIN tender ON bid.tender_id = tender.id
+				WHERE tender.buyer_id = $1 AND bid.tender_id = $2
+				LIMIT $3 OFFSET $4
+			`;
+			bidsParams = [userId, tenderID, itemsPerPage, offset];
+		} else {
+			return res.status(403).json({ code: "FORBIDDEN" });
```
Comment (soonick) [inline server/api.js:428]: We should show something for admins, but we can take care of that later.

File: server/api.js [hunk 2]
```diff
@@ -390,32 +391,62 @@ router.get("/tenders", async (req, res) => {
...
-		[tenderID, itemsPerPage, offset]
-	);
+		const totalBidsResult = await db.query(totalBidsQuery, totalBidsParams);
+		const totalPages = Math.ceil(totalBidsResult.rows[0].count / itemsPerPage);
+		const bidsResult = await db.query(bidsQuery, bidsParams);
 
-	totalBidsResults
-		? res.send({
-				results: totalBidsResults.rows,
+		if (bidsResult) {
```
Comment (soonick) [inline server/api.js]: Remove the if statement. No bids in the DB is not a server error, it just means that results will be an empty array.

**Review by soonick — CHANGES_REQUESTED** [review]

**Review by soonick — APPROVED** [review]

**Pull request merged**